src/utils/id.ts

```diff
@@ -15,3 +17,16 @@ export function generateUniqueId(byteCount: number): string {
   }
   return String.fromCharCode.apply(null, SHARED_CHAR_CODES_ARRAY.slice(0, byteCount * 2));
 }
+
+export function generateSessionId(): string {
+  // if the browser does support local storage the session is tracked over multiple page loads
+  const sessionFlags = localStorage.isSupported ? "00" : "01";
+  return `${sessionFlags}${generateUniqueId(SESSION_ID_BYTES - 1)}`;
+}
+
+export function generateTraceId(sessionId: string | null): string {
+  if (!sessionId) {
+    return `d04201${generateUniqueId(TRACE_ID_BYTES - 3)}`;
+  }
+  return `d04200${sessionId}${generateUniqueId(TRACE_ID_BYTES - SESSION_ID_BYTES - 3)}`;
+}
```

Comment:
Suggestion: lets somehow make the generation of the flag values very explicit so its easy to trace which condition results in which flags being set.
We could also consider doing the same for the traceparent flags